docs/DeveloperGuide.md

```diff
@@ -421,20 +433,20 @@ Lastly, the `MacroRunner` utility class in the `Logic` component is responsible
 #### 7.5.3 Execution of macros
 
 Here we detail what exactly happens in each macro's `toCommandExecutable` object.
-Throughout this section, we will use the following macro as an example: `macro test a;add -n \a breakfast 200`.
+Throughout this section, we will use the following macro as an example: `macro test a;add -n \a breakfast -c 200`.
 
-1. Parameters supplied by the user are substituted into the macro's lines using regular expressions. For the example if the user enters `test -a rice`, `rice` will be substituted into the `\a` part of `add -n \a breakfast 200` to obtain `add -n rice breakfast 200`;
+1. Parameters supplied by the user are substituted into the macro's lines using regular expressions. For the example if the user enters `test -a rice`, `rice` will be substituted into the `\a` part of `add -n \a breakfast -c 200` to obtain `add -n rice breakfast -c 200`;
 2. Pass the list of strings to a new instance of `PrimitiveCommandParser` to get a list of `Command`s. If a `ParseException` occurs, we recast it as a `CommandException` and re-throw it.
 3. Execute all these `Command`s in sequence, and return a new `CommandResult` created from concatenating the respective `CommandResults` returned by the `Command`s. If a `CommandException` occurs in this sequence, that exception is thrown with the added message listing the commands that have executed successfully, and commands that have yet to be executed.
 
 Below is a sequence diagram summarizing the main interactions.
 
 ![Sequence diagram for macro](images/MacroSequenceDiagram.png)
 
-#### 7.5.3 Additional considerations for this implementation
-Alternatively could have done the parsing/compiling on creation of the macro so we don't need to use another parser during execution.
-That is, each macro instance could contain its own `ParameterSet` which could then be used by its callee functions.
-This implementation is considerably more involved hence we proceeded with the current implementation.
+##### 7.5.4 Aspect: How undo executes
+
+* **Alternative 1:** done the parsing/compiling on creation of the macro so we don't need to use another parser during execution.
+  * Cons: More involved and prone to bugs
```

Comment:
We shud add the current implementation similar to other Aspect.

And it should be Additional considerations for this implementation instead of how undo executes